src/object_accessor.hpp

```diff
@@ -128,91 +94,75 @@ struct MutationOutsideTransactionException : public std::logic_error {
         if (property.type == PropertyType::Object) {
-            m_row.nullify_link(column);
+            if (!is_default)
+                table.nullify_link(column, row);
         }
         else {
-            m_row.set_null(column);
+            table.set_null(column, row, is_default);
         }
         return;
     }
 
     switch (property.type) {
         case PropertyType::Bool:
-            m_row.set_bool(column, Accessor::to_bool(ctx, value));
+            table.set_bool(column, row, Accessor::to_bool(ctx, value), is_default);
             break;
         case PropertyType::Int:
-            if (property.is_primary)
-                m_row.set_int_unique(column, Accessor::to_long(ctx, value));
-            else
-                m_row.set_int(column, Accessor::to_long(ctx, value));
+            table.set_int(column, row, Accessor::to_long(ctx, value), is_default);
             break;
         case PropertyType::Float:
-            m_row.set_float(column, Accessor::to_float(ctx, value));
+            table.set_float(column, row, Accessor::to_float(ctx, value), is_default);
             break;
         case PropertyType::Double:
-            m_row.set_double(column, Accessor::to_double(ctx, value));
+            table.set_double(column, row, Accessor::to_double(ctx, value), is_default);
             break;
         case PropertyType::String: {
-            auto string_value = Accessor::to_string(ctx, value);
-            if (property.is_primary)
-                m_row.set_string_unique(column, string_value);
-            else
-                m_row.set_string(column, string_value);
+            auto str = Accessor::to_string(ctx, value);
+            table.set_string(column, row, str, is_default);
             break;
         }
-        case PropertyType::Data:
-            m_row.set_binary(column, BinaryData(Accessor::to_binary(ctx, value)));
+        case PropertyType::Data: {
+            auto data = Accessor::to_binary(ctx, value);
+            table.set_binary(column, row, BinaryData(data), is_default);
             break;
+        }
         case PropertyType::Any:
-            m_row.set_mixed(column, Accessor::to_mixed(ctx, value));
+            table.set_mixed(column, row, Accessor::to_mixed(ctx, value), is_default);
             break;
         case PropertyType::Date:
-            m_row.set_timestamp(column, Accessor::to_timestamp(ctx, value));
+            table.set_timestamp(column, row, Accessor::to_timestamp(ctx, value), is_default);
             break;
         case PropertyType::Object: {
-            if (Accessor::is_null(ctx, value)) {
-                m_row.nullify_link(column);
-            }
-            else {
-                m_row.set_link(column, Accessor::to_object_index(ctx, m_realm, value, property.object_type, try_update));
-            }
+            table.set_link(column, row, Accessor::to_object_index(ctx, m_realm, value, property.object_type, try_update), is_default);
```

Comment:
`value` can't be null here (there's a check for that before the switch statement).